### DISCUSSION TITLE
What is best way to do equal check on 2 TypeDefinitionRegistry objects or 2 ObjectTypeDefinition objects?

### DISCUSSION THREAD

**Discussion by arjunballa** [graphql-java/graphql-java#2537] [Q&A]
I have a requirement to compare 2 graphql api schema definitions for equality. 
I looked at `objectTypeDefinition.isEqualTo(Node)` method but it is just doing name comparison but not checking for `fieldDefinitions`

What is the best way to do?

**Comment by bbakerman** [comment]
The first thing to recognise is that the `XXXDefinition` objects are parsed AST nodes - they are not types.  They are parsed bits of text.

And yes the equality operations on them (including the `.equalTo` methods) have not been implemented.  We do not recommend using them for equality checks.  

But again they are NOT graphql types - they are parsed elements from SDL text

**Reply by arjunballa** [reply]
@bbakerman Thanks for the reply. I already noticed that but even `GraphQLSchema` which I assume is representation of DL which also have lot of validations does not implement .equals.

How do I compare 2 graphql DLs? Any pointers?

**Reply by bbakerman** [reply]
> I already noticed that but even GraphQLSchema which I assume is representation of DL which also have lot of validations does not implement .equals.

Ahh the reason for that is that the types are designed as Singletons - within a schema say there is only 1 `GraphqlObjectType` in existence.  hence we can do direct `==` compares.

In short we dont have `.equals` support on types (are they equal of they have the same directives right down to the argument level say??) and you would have to write your own
